src/officehours/routing.py

```diff
@@ -1,12 +1,22 @@
 from channels.auth import AuthMiddlewareStack
 from channels.routing import ProtocolTypeRouter, URLRouter
+from channels.security.websocket import AllowedHostsOriginValidator
+
+from django.core.asgi import get_asgi_application
+
 import officehours_api.routing
 
+
+asgi_application = get_asgi_application()
+
 application = ProtocolTypeRouter({
     # (http->django views is added by default)
-    'websocket': AuthMiddlewareStack(
-        URLRouter(
-            officehours_api.routing.websocket_urlpatterns
+    'http': asgi_application,
+    'websocket': AllowedHostsOriginValidator(
```

Comment:
Was this just done for additional security or was there some other reason here?

Reply:
Yeah, I saw it in some of the Channels tutorials and wondered why it wasn't included here. Seemed like an easy thing to add. Here's the documentation on it: https://channels.readthedocs.io/en/stable/topics/security.html